### PR TITLE
fix: Make sidebar item to indicate the active page gf-118

### PR DESCRIPTION
**Pull request by HoroshkoMykhailo**
Summary

- The active page is highlighted on the sidebar when the page is opened.
- When hovering over sidebar item the background changes to gray.

image

image

### CODE REVIEW

File: apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
-			<label className={styles["navigation-item-label"]}>{label}</label>
-		</NavLink>
-	</li>
-);
+const SidebarItem = ({ href, icon, label }: NavigationItem): JSX.Element => {
+	const location = useLocation();
+	const currentPath = location.pathname;
+
+	const isActive = currentPath === href;
+
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx:13]: As I remember NavLink already has isActive flag when adding styles
```suggestion
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/sidebar/sidebar-item/helper/get-class-name-nav.helper.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import styles from "../styles.module.css";
+
+const getClassName = ({ isActive }: { isActive: boolean }): string => {
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/sidebar/sidebar-item/helper/get-class-name-nav.helper.ts:3]: let's keep this helper inside SidebarItem component, it most likely will depend on other props later, and usually we keep className related logic inside components

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx [hunk 3]
```diff
@@ -5,15 +5,21 @@ import { type NavigationItem } from "~/libs/types/types.js";
...
-	</li>
-);
+const getClassName = ({ isActive }: { isActive: boolean }): string => {
+	return `${styles["navigation-link"] ?? ""} ${isActive ? (styles["active"] ?? "") : ""}`;
+};
+
+const SidebarItem = ({ href, icon, label }: NavigationItem): JSX.Element => {
+	return (
+		<li className={styles["sidebar-item"]}>
+			<NavLink className={getClassName} to={href}>
```
Comment (what1s1ove) [inline apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx:15]: Remove all `import { NavLink } from "react-router-dom";` from the project
we need to use our `Link` component wrapper. You can use `NavLink` from `react-router-dom` in our Link component

Comment (HoroshkoMykhailo) [reply apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx:15]: But how then change style of Link component? It was said that we cannot pass classname to Link component

File: apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx [hunk 4]
```diff
@@ -5,15 +5,21 @@ import { type NavigationItem } from "~/libs/types/types.js";
...
-			<span className={styles["navigation-icon"]}>
-				<Icon height={20} name={icon} width={20} />
-			</span>
-			<label className={styles["navigation-item-label"]}>{label}</label>
-		</NavLink>
-	</li>
-);
+const getClassName = ({ isActive }: { isActive: boolean }): string => {
+	return `${styles["navigation-link"] ?? ""} ${isActive ? (styles["active"] ?? "") : ""}`;
+};
```
Comment (perekotypole) [inline apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx:10]: Might it better to use `get-valid-classes` helper?

File: apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx (hunk 3 above)
Comment (liza-veis) [reply apps/frontend/src/libs/components/sidebar/sidebar-item/sidebar-item.tsx:17]: Let's leave it as it is in this PR, will create a separate ticket when we decide on approach

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]

**Review by yevhenii-hladkov — APPROVED** [review]

**Pull request merged**